app/assets/javascripts/services/topology_service.js

```diff
@@ -216,26 +217,20 @@ ManageIQ.angular.app.service('topologyService', function() {
   // this injects some common code in the controller - temporary pending a proper merge
   this.mixinSearch = function($scope) {
     var topologyService = this;
-
+      // NOTE: listener on search
+    ManageIQ.angular.rxSubject.subscribe(function(event) {
+      if (event.service === 'topologyService') {
+        var query = $('input#search_topology')[0].value;
+        var svg = topologyService.getSVG($scope.d3);
+        if (event.name === 'searchNode') {
+          topologyService.searchNode(svg, query);
+        } else if (event.name === 'resetSearch') {
+          topologyService.resetSearch($scope.d3);
+          $('input#search_topology')[0].value = '';
```

Comment:
Looks like you're dropping the code setting `searching` and `notFound` .. looks like a bug, that code is definitely used ;).

Also, you need to update the `app/views/shared/topology/_not_found.html.haml` if you move `searching` and `notFound` to `vm`. (But you probably can't do that until all the topologies are converted.)